### PR TITLE
[exiv2] Make add nls feature

### PR DESCRIPTION
**Pull request by m-kuhn**
**Describe the pull request**

- optional support for nls

### CODE REVIEW

File: ports/exiv2/portfile.cmake [hunk 1]
```diff
@@ -53,6 +53,11 @@ vcpkg_install_cmake()
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/exiv2)
 vcpkg_fixup_pkgconfig()
 
+if("nls" IN_LIST FEATURES)
+    set(EXIV2_ENABLE_NLS ON)
```
Comment (BillyONeal) [inline ports/exiv2/portfile.cmake:57]: This doesn't seem connected to anything? I think you need to move it up and change

`-DEXIV2_ENABLE_NLS=OFF` 
to
`-DEXIV2_ENABLE_NLS=${EXIV2_ENABLE_NLS}`

Comment (BillyONeal) [reply ports/exiv2/portfile.cmake:57]: Actually I note that this if is already the behavior of the vcpkg_check_features above.

**Review by BillyONeal — CHANGES_REQUESTED** [review]
The feature doesn't seem to be connected.

File: versions/e-/exiv2.json [hunk 2]
```diff
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "f716a33e59b24c79140216d836644fe0603e322b",
+      "git-tree": "1e93f266421999dd82b931b2dbf672d255b14b43",
+      "version": "0.27.4",
+      "port-version": 1
+    },
+    {
+      "git-tree": "7332f311b24480f90033a1207893c6f8f577b0e3",
```
Comment (JonLiu1993) [inline versions/e-/exiv2.json:9]: Why replace the version commit of the previous version?

File: versions/e-/exiv2.json [hunk 3]
```diff
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "f716a33e59b24c79140216d836644fe0603e322b",
```
Comment (JonLiu1993) [inline versions/e-/exiv2.json:4]: Why replace the version commit of the previous version?

Comment (m-kuhn) [reply versions/e-/exiv2.json:4]: Doesn't this need a new port-version? I'll follow your lead here...

File: ports/exiv2/portfile.cmake (hunk 1 above)
Comment (m-kuhn) [reply ports/exiv2/portfile.cmake:57]: To make it available for configure file:
https://github.com/microsoft/vcpkg/pull/18964/files#diff-1bdf222c873035319124c25b484ba3840ebc8adb43ea4731e43e4627dcf39ac3L56

Being used in here:
https://github.com/microsoft/vcpkg/pull/18964/files#diff-2f0b76215a2d1a149da5bfea386e4e22c09569045184d1b9fbd363fd4b784e5aR5

Is that not the correct approach?

File: versions/e-/exiv2.json (hunk 3 above)
Comment (JonLiu1993) [reply versions/e-/exiv2.json:4]: Yes, it is necessary, but we should‘t change the commit of the previous version

File: versions/e-/exiv2.json (hunk 2 above)
Comment (JonLiu1993) [inline versions/e-/exiv2.json:9]:
```suggestion
      "git-tree": "f716a33e59b24c79140216d836644fe0603e322b",
```

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/exiv2/portfile.cmake (hunk 1 above)
Comment (BillyONeal) [reply ports/exiv2/portfile.cmake:57]: Ah, I see. That makes sense, thanks! I also missed that there is a ${FEATURE_OPTIONS} for the other features so I was mistaken about that too -- thanks GitHub for hiding important context in your code review tool :(

**Review by BillyONeal — APPROVED** [review]

File: ports/exiv2/portfile.cmake (hunk 1 above)
Comment (m-kuhn) [reply ports/exiv2/portfile.cmake:57]: > thanks GitHub for hiding important context in your code review tool :(

deciding what's relevant to get the best signal to noise ratio sounds like a non-trivial job ...

but while we are at it, I think this repository could benefit from some ci tooling to detect things like incorrect or missing usage of `x-add-version` and ideally even some integrated bot magic for fixing these issues...

Comment (BillyONeal) [reply ports/exiv2/portfile.cmake:57]: >deciding what's relevant to get the best signal to noise ratio sounds like a non-trivial job ...

The Azure DevOps code review tool just always shows you the whole file, it's never confusing. But there's no way to configure GitHub to do that :(

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**